test/e2e_node/dra_test.go

```diff
@@ -66,14 +72,19 @@ var _ = framework.SIGDescribe("node")("DRA", feature.DynamicResourceAllocation,
 	f := framework.NewDefaultFramework("dra-node")
 	f.NamespacePodSecurityLevel = admissionapi.LevelBaseline
 
-	var kubeletPlugin, kubeletPlugin1, kubeletPlugin2 *testdriver.ExamplePlugin
-
-	f.Context("Resource Kubelet Plugin", f.WithSerial(), func() {
-		ginkgo.BeforeEach(func(ctx context.Context) {
-			kubeletPlugin = newKubeletPlugin(ctx, getNodeName(ctx, f), driverName)
+	ginkgo.BeforeEach(func() {
+		ginkgo.DeferCleanup(func(ctx context.Context) {
```

Comment:
```suggestion
	ginkgo.AfterEach(func(ctx context.Context) {
```

Reply:
No, `ginkgo.AfterEach` has a different semantic. Use it to run additional, common checks after a set of tests (like checking some post-condition), but not for cleanup.

Reply:
From https://onsi.github.io/ginkgo/#spec-cleanup-aftereach-and-defercleanup
> ... AfterEach and JustAfterEach. These are used to clean up after specs

What you said is not the common semantic, IMO.

Reply:
What you link to seems to have been written before `DeferCleanup` was added.

@onsi: care to comment?

Reply:
See also: https://github.com/onsi/ginkgo/blob/master/docs/index.md#cleaning-up-our-cleanup-code-defercleanup

Reply:
I think the key message is "bring our intended cleanup closer to our setup code and avoid extracting a separate closure variable". Since we don't have setup code or closure variable in this case, I would personally just use the shorter and simpler form. Anyway, I think this is not very important.

Reply:
The order of callback invocation is also different (AfterEach = first-in-first-out vs. DeferCleanup = last-in-first-out as with `defer` or `testing.Cleanup`). We want the latter and thus what the code is using right now. What you suggest would change the order in which the different cleanup steps run.

Reply:
hey all - both will work.  I'm preferring `DeferCleanup` more and more and may update the docs (which span a wide range of features and time at this point) to reflect that shift.  It's much easier to reason about the behavior of `DeferCleanup`

with that said - the `AfterEach` do honor the container nesting - so it is deepest-in is first-out.  But `DeferCleanup` makes this so much cleaner ("if this `BeforeEach` ran then the `DeferCleanup` is agreed to run on lifo order).